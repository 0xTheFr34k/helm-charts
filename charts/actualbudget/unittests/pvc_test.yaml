suite: test persistent volume claim

templates:
  - pvc.yaml

release:
  name: actualbudget
  namespace: actualbudget

chart:
  version: 1.0.0
  appVersion: 1.0.0

tests:
  - it: should not create pvc when persistence is not enabled
    set:
      persistence:
        enabled: false
    asserts:
      - hasDocuments:
          count: 0

  - it: should not create pvc when existingClaim is set
    set:
      persistence:
        existingClaim: fake-existing-claim
    asserts:
      - hasDocuments:
          count: 0

  - it: should set annotations when persistence is enabled and existingClaim is not set and annotations are set
    set:
      persistence:
        enabled: true
        annotations:
          key: value
        existingClaim: ""
        size: 20Gi
    asserts:
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations.key
          value: value
      - equal:
          path: metadata.name
          value: actualbudget-data
      - equal:
          path: spec.resources.requests.storage
          value: 20Gi

  - it: should set volumeMode when persistence is enabled and volumeMode is set
    set:
      persistence:
        enabled: true
        volumeMode: Block
    asserts:
      - equal:
          path: spec.volumeMode
          value: Block

  - it: should set accessModes when persistence is enabled and accessModes are set
    set:
      persistence:
        enabled: true
        accessModes:
          - ReadWriteOnce
    asserts:
      - hasDocuments:
          count: 1
      - contains:
          path: spec.accessModes
          content: ReadWriteOnce

  - it: should set storageClass when persistence is enabled and storageClass is set
    set:
      persistence:
        enabled: true
        storageClass: "longhorn-static"
    asserts:
      - equal:
          path: spec.storageClassName
          value: longhorn-static

  - it: should match snapshot of default values
    set:
      persistence:
        enabled: true
        storageClass: "longhorn-static"
        volumeMode: Filesystem
    asserts:
      - matchSnapshot: { }
