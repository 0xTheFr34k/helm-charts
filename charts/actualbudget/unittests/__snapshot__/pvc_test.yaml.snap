should match snapshot of default values:
  1: |
    apiVersion: v1
    kind: PersistentVolumeClaim
    metadata:
      labels:
        app.kubernetes.io/instance: actualbudget
        app.kubernetes.io/managed-by: Helm
        app.kubernetes.io/name: actualbudget
        app.kubernetes.io/version: 1.0.0
        helm.sh/chart: actualbudget-1.0.0
      name: actualbudget-data
      namespace: actualbudget
    spec:
      accessModes:
        - ReadWriteOnce
      resources:
        requests:
          storage: 10Gi
      storageClassName: longhorn-static
      volumeMode: Filesystem
